### ISSUE TITLE
Populate NetBox inventory via active discovery

### ISSUE THREAD

**Issue by mmguero** [cisagov/Malcolm#477]
**[<img src="https://avatars.githubusercontent.com/u/13872653?s=17&v=4" width="17" height="17"> @mmguero](https://github.com/mmguero)** cloned issue [idaholab/Malcolm#136](https://github.com/idaholab/Malcolm/issues/136) on 2022-12-07: 

> Feature-tracking issue dependent on #131

**Comment by mmguero** [comment]
**[<img src="https://avatars.githubusercontent.com/u/13872653?s=17&v=4" width="17" height="17"> @mmguero](https://github.com/mmguero)** [commented](https://github.com/idaholab/Malcolm/issues/136#issuecomment-1375676947) on 2023-01-09: 

> Things to look at:
> 
> 
> 
> * mDNS
> 
> * uPnP
> 
> * ping
> 
> * snmp
> 
> * arp broadcasts
> 
> * nmap-style port scanning
> 
> * CDP (cisco discovery protocol)
> 
> * Link Layer Discovery Protocol (LLDP)
> 
> * SCTP INIT packets
> 
> * [netmiko](https://github.com/ktbyers/netmiko)
> 
> * [IX-DiscoveryTools](https://github.com/idaholab/IX-DiscoveryTools)
> 
> * [netbox-agent](https://github.com/Solvik/netbox-agent) (though I don't think we want to do agent-based)
> 
> 
> 
> Some resources
> 
> * GitHub [topics/host-discovery](https://github.com/topics/host-discovery)

**Comment by mmguero** [comment]
**[<img src="https://avatars.githubusercontent.com/u/13872653?s=17&v=4" width="17" height="17"> @mmguero](https://github.com/mmguero)** [commented](https://github.com/idaholab/Malcolm/issues/136#issuecomment-1549939057) on 2023-05-16: 

> As part of our collaboration with BSI-Bund ([www.bsi.bund.de](https://www.bsi.bund.de/EN/Home/home_node.html), [github org](https://github.com/orgs/BSI-Bund/)) with work being worked on by [Fraunhofer IOSB](https://www.iosb.fraunhofer.de/en.html), they are going to be taking over this development of this feature. Marking as external, will update this issue as their work continues.

**Comment by mmguero** [comment]
**[<img src="https://avatars.githubusercontent.com/u/13872653?s=17&v=4" width="17" height="17"> @mmguero](https://github.com/mmguero)** [commented](https://github.com/idaholab/Malcolm/issues/136#issuecomment-1781991146) on 2023-10-26: 

> Another thing to look into: https://github.com/dainok/netdoc

**Comment by Crubumble** [comment]
Please note that this feature is postponed since the plugins for populate the NetBox database and the vulnerability handling have higher priority.
